src/main/java/org/jenkinsci/plugins/awsdevicefarm/AWSDeviceFarm.java

```diff
@@ -279,14 +283,33 @@ public VPCEConfiguration getVPCEConfiguration(String vpceServiceName) throws AWS
         List<Upload> testSpecUploads = new ArrayList<Upload>();
         for (Upload upload : allUploads) {
 			if (upload.getType().contains("TEST_SPEC")
-					&& UploadStatus.SUCCEEDED.toString().equals(upload.getStatus())) {
+					&& UploadStatus.SUCCEEDED.toString().equals(upload.getStatus()) && !isRestrictedDefaultSpec(upload)) {
 				testSpecUploads.add(upload);
 
 			}
         }
         return testSpecUploads;
     }
 
+    /**
+     * Helper function to detect a default testspec file for frameworks that cannot use it.
+     *
+     * @param upload Testspec
+     *
+     * @return true if it is a default spec file.
+     * @throws AWSDeviceFarmException
+     */
+    public Boolean isRestrictedDefaultSpec(Upload testSpec) {
+        if ((testSpec.getType().equals("APPIUM_RUBY_TEST_SPEC") ||
```

Comment:
Reverse the equals statement since this can lead to null pointer exception
APPIUM_RUBY_TEST_SPEC.equals(testspec.gettype())